modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMFormValuesToMapConverterImpl.java

```diff
@@ -61,84 +60,29 @@
 			return Collections.emptyMap();
 		}
 
-		DDMForm ddmForm = ddmStructure.getDDMForm();
-
-		_addMissingDDMFormFieldValues(
-			ddmForm.getDDMFormFields(), ddmFormValues);
-
-		Map<String, DDMFormField> ddmFormFields =
+		Map<String, DDMFormField> ddmFormFieldsMap =
 			ddmStructure.getFullHierarchyDDMFormFieldsMap(true);
 
-		Map<String, Object> values = new LinkedHashMap<>(ddmFormFields.size());
-
-		for (DDMFormFieldValue ddmFormFieldValue :
-				ddmFormValues.getDDMFormFieldValues()) {
-
-			if (ddmFormFields.containsKey(ddmFormFieldValue.getName())) {
-				_addValues(ddmFormFields, ddmFormFieldValue, values);
-			}
-		}
-
-		return values;
-	}
-
-	private void _addMissingDDMFormFieldValues(
-		List<DDMFormField> ddmFormFields, DDMFormValues ddmFormValues) {
+		Map<String, Object> values = new LinkedHashMap<>(
+			ddmFormFieldsMap.size());
 
-		Map<String, List<DDMFormFieldValue>> ddmFormFieldValues =
-			ddmFormValues.getDDMFormFieldValuesMap(false);
+		List<DDMFormFieldValue> ddmFormFieldValues =
+			DDMFormValuesConverterUtil.addMissingDDMFormFieldValues(
+				ddmFormFieldsMap.values(),
+				ddmFormValues.getDDMFormFieldValuesMap(true));
 
-		for (DDMFormField ddmFormField : ddmFormFields) {
-			if (!ddmFormFieldValues.containsKey(ddmFormField.getName()) &&
-				!GetterUtil.getBoolean(
-					ddmFormField.getProperty("upgradedStructure"))) {
+		Stream<DDMFormFieldValue> ddmFormFieldValuesStream =
+			ddmFormFieldValues.stream();
 
-				DDMFormFieldValue ddmFormFieldValue = new DDMFormFieldValue() {
-					{
-						setInstanceId(StringUtil.randomString());
-						setName(ddmFormField.getName());
-					}
-				};
+		ddmFormFieldValuesStream.filter(
+			ddmFormFieldValue -> ddmFormFieldsMap.containsKey(
+				ddmFormFieldValue.getName())
+		).forEach(
+			ddmFormFieldValue -> _addValues(
```

Comment:
@mateussgn, I realized that the first parameter of the method `_addValues` is named as `ddmFormFields` though its type is a `Map`, can you please rename it to `ddmFormFieldsMap` to be consistent with your new code?

Reply:
Sure! Thank you for noticing it!